### PR TITLE
Save properties of a test case

### PR DESCRIPTION
**Pull request by fkorotkov**
Some reporters use them to store additional information like file location

### CODE REVIEW

**Comment by fkorotkov** [comment]
@joshdk PTAL. 🙌

File: ingest_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+func TestInjest(t *testing.T) {
+	tests := []struct {
+		title    string
+		input    []byte
+		expected string
+	}{
+		{
+			title: "xml input",
+			input: []byte(`<testsuite errors="0" failures="1" file="Foo.java"><testcase name="unit tests" file="Foo.java"/></testsuite>`),
+			expected: `
```
Comment (joshdk) [inline ingest_test.go]: Have this be a slice of Suite literals.

File: ingest_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+]`,
+		},
+	}
+
+	for index, test := range tests {
+		name := fmt.Sprintf("#%d - %s", index+1, test.title)
+
+		t.Run(name, func(t *testing.T) {
+			actual, err := Ingest(test.input)
+			if err != nil {
```
Comment (joshdk) [inline ingest_test.go]: Use [`github.com/stretchr/testify/require.Nil()`](https://godoc.org/github.com/stretchr/testify/require#Nil) here.

File: ingest_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+				t.Fatal(err)
+			}
+			if actual == nil {
```
Comment (joshdk) [inline ingest_test.go]: You can use [`github.com/stretchr/testify/require.NotEmpty`](https://godoc.org/github.com/stretchr/testify/require#NotEmpty) for this check.

File: ingest_test.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+				t.Fatalf("No suites found!")
+			}
+
+			actualJSON, err := json.MarshalIndent(actual, "", "  ")
```
Comment (joshdk) [inline ingest_test.go]: Modify `expected` to be a slice of `Suite` literals, and compare data-to-data, not need to marshal back into json. Maintenance (and writing more tests) will then be easier in the future 🙂

File: types.go [hunk 5]
```diff
@@ -127,6 +127,10 @@ type Test struct {
 	//   Error == nil && (Status == Passed || Status == Skipped)
 	//   Error != nil && (Status == Failed || Status == Error)
 	Error error `json:"error" yaml:"error"`
+
+	// Additional properties from XML node attributes.
+	// Some tools use them to store additional information about test location.
+	Properties map[string]string
```
Comment (joshdk) [inline types.go]:
```suggestion
	Properties map[string]string `json:"properties" yaml:"properties"`
```

**Comment by fkorotkov** [comment]
Thank you for the review, @joshdk! Addressed your comments. PTAL 🙌

**Pull request merged**